gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/test/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAEnrollEndpointTest.java

```diff
@@ -675,4 +725,157 @@ public void shouldAcceptEnrollment_MFAForceEnrollment() throws Exception {
                 },
                 HttpStatusCode.FOUND_302, "Found", null);
     }
+
+    @Test
+    public void shouldEnrollWhenUserCannotSkip() throws Exception {
```

Comment:
**todo**: not sure the name of this test reflect the purpose of this test, by reading the name my expectation is the user has to be redirected to the enrollment page because he can't skip the enrollment but here the user already has an ACTIVATED factor so the enrollment is ignored and the user is redirected to /authorize